File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        
        for (int move : moves) {
            
            int doll = pickDoll(board, move);
            if (doll == -1) {
                continue;
            }
            // System.out.format("doll : %s %n", doll);
            // Arrays.stream(board)
            //     .forEach(e->{
            //         System.out.println(Arrays.toString(e));
            //     });
            // System.out.println();
            if (!stack.empty()  && stack.peek() == doll) {
                stack.pop();
                answer += 2;
            }
            else {
                stack.push(doll);
            }
        }   
                
        return answer;
    }
    
    public int pickDoll(int[][] board, int colIdx) {
        int colLength = board[0].length;
        int rowLength = board.length;
        colIdx -= 1;
        int answer = 0;
        for (int rowIdx=0; rowIdx < rowLength; rowIdx++) {
            
            if (board[rowIdx][colIdx] != 0) {
                answer = board[rowIdx][colIdx];
                board[rowIdx][colIdx] = 0;
                return answer;
            }
        }
        return -1;
    }
    
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
from collections import *
def solution(board, moves):
    answer = 0
    stack =  []
    length = len(board)
    for k in board :
        print(k)
    print()
    
    for move in moves:
        
        i = move - 1
        
        for j in range(0, length):
            # print(f'{i},{j}, board[{j}][{i}] : {board[j][i]}')
            # print(f'시작 전 stack : {stack}')
            if board[j][i] != 0:
                value = board[j][i]
                if stack and stack[-1] == value :
                    stack.pop()
                    answer += 2
                else: 
                    stack.append(value)
                board[j][i] = 0
                break
        else:
            continue        
        
        # for k in board :
            # print(k)
        
        # print(f'moving 후 stack : {stack}')
        # if stack and stack[-1] == value :
            # stack.pop()
            # answer += 2
        # print(stack)
        # print()
        
    
    return answer